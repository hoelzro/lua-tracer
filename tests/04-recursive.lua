local function fib(n)
  assert(n >= 1)

  if n == 1 or n == 2 then
    return 1
  end

  return fib(n - 2) + fib(n - 1) -- trace: n
end

fib(10)
