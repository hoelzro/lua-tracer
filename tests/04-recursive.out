n	10
n	8
n	6
n	4
n	3
n	5
n	3
n	4
n	3
n	7
n	5
n	3
n	4
n	3
n	6
n	4
n	3
n	5
n	3
n	4
n	3
n	9
n	7
n	5
n	3
n	4
n	3
n	6
n	4
n	3
n	5
n	3
n	4
n	3
n	8
n	6
n	4
n	3
n	5
n	3
n	4
n	3
n	7
n	5
n	3
n	4
n	3
n	6
n	4
n	3
n	5
n	3
n	4
n	3
